shell/platform/darwin/ios/framework/Source/connection_collection.h

```diff
@@ -0,0 +1,30 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SHELL_PLATFORM_IOS_FRAMEWORK_SOURCE_CONNECTION_COLLECTION_H_
+#define SHELL_PLATFORM_IOS_FRAMEWORK_SOURCE_CONNECTION_COLLECTION_H_
+
+#include <cstdint>
+#include <map>
+#include <string>
+
+namespace flutter {
+
+/// Maintains a current integer assigned to a name (connections).
+class ConnectionCollection {
+ public:
+  typedef int64_t Connection;
```

Comment:
nit: ConnectionCounter?

Reply:
also, docstring for this

Reply:
From an API perspective it is just a token that's returned for cleanup, the fact it is a number shouldn't be important.  Thinking of it as anything other than a connection would be confusing.  

If you want to get technical everything in software is just a number that has meaning through artifice, but we don't call everything numbers, we call them by their abstractions =)

Reply:
But you compare it to an integer later outside of this class - if you really want this to be an enum of some sort, maybe use an enum?

Reply:
Sure, I made some helper functions and a const so people don't ever have to care that it is an integer.  There still is code that uses that knowledge in the channels though to avoid bringing the code into macOS without need.